sktree/tree/_honest_tree.py

```diff
@@ -0,0 +1,338 @@
+        Names of features seen during :term:`fit`. Defined only when `X`
+        has feature names that are all strings.
+
+        .. versionadded:: 1.0
+
+    n_outputs_ : int
+        The number of outputs when ``fit`` is performed.
+
+    tree_ : Tree instance
+        The underlying Tree object. Please refer to
+        ``help(sklearn.tree._tree.Tree)`` for attributes of Tree object and
+        :ref:`sphx_glr_auto_examples_tree_plot_unveil_tree_structure.py`
+        for basic usage of these attributes.
+
+    empirical_prior_ : float
+        Proportion of each class in the training labels y
+
+    structure_indices_ : numpy.ndarray, shape=(n_structure,)
+        Indices of training samples used to learn the structure
+
+    honest_indices_ : numpy.ndarray, shape=(n_honest,)
+        Indices of training samples used to learn leaf estimates
+
+    Notes
+    -----
+    The default values for the parameters controlling the size of the trees
+    (e.g. ``max_depth``, ``min_samples_leaf``, etc.) lead to fully grown and
+    unpruned trees which can potentially be very large on some data sets. To
+    reduce memory consumption, the complexity and size of the trees should be
+    controlled by setting those parameter values.
+
+    The :meth:`predict` method operates using the :func:`numpy.argmax`
+    function on the outputs of :meth:`predict_proba`. This means that in
+    case the highest predicted probabilities are tied, the classifier will
+    predict the tied class with the lowest index in :term:`classes_`.
+
+    References
+    ----------
+
+    .. [1] https://en.wikipedia.org/wiki/Decision_tree_learning
+
+    .. [2] L. Breiman, J. Friedman, R. Olshen, and C. Stone, "Classification
+            and Regression Trees", Wadsworth, Belmont, CA, 1984.
+
+    .. [3] T. Hastie, R. Tibshirani and J. Friedman. "Elements of Statistical
+            Learning", Springer, 2009.
+
+    .. [4] L. Breiman, and A. Cutler, "Random Forests",
+            https://www.stat.berkeley.edu/~breiman/RandomForests/cc_home.htm
+
+    .. [5] S. Athey, J. Tibshirani, and S. Wager. "Generalized
+            Random Forests", Annals of Statistics, 2019.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.model_selection import cross_val_score
+    >>> from honest_forests import HonestTreeClassifier
+    >>> clf = HonestTreeClassifier(random_state=0)
+    >>> iris = load_iris()
+    >>> cross_val_score(clf, iris.data, iris.target, cv=10)
+    ...                             # doctest: +SKIP
+    ...
+    array([0.93333333, 0.93333333, 1.        , 1.        , 0.93333333,
+           0.8       , 0.8       , 0.93333333, 1.        , 1.        ])
+    """
+
+    def __init__(
+        self,
+        estimator=None,
+        honest_fraction=0.5,
+        honest_prior="empirical",
+    ):
+        if not estimator:
+            self.estimator_ = DecisionTreeClassifier()
+        self.estimator_ = estimator
```

Comment:
You can migrate this to within the `fit` function and then set `self.estimator=estimator`.